plugins/example-todo-list/src/components/TodoListPage/TodoListPage.tsx

```diff
@@ -0,0 +1,187 @@
+  const [editElement, setEdit] = useState<Todo | undefined>();
+
+  const handleAdd = async () => {
+    try {
+      const response = await fetch(
+        `${await discoveryApi.getBaseUrl('todolist')}/todos`,
+        {
+          method: 'POST',
+          body: JSON.stringify({ title: title.current }),
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        },
+      );
+      if (!response.ok) {
+        const { error } = await response.json();
+        alertApi.post({
+          message: error.message,
+          severity: 'error',
+        });
+        return;
+      }
+      refetchTodos();
+    } catch (e: any) {
+      alertApi.post({ message: e.message, severity: 'error' });
+    }
+  };
+
+  const handleEdit = async (todo: Todo) => {
+    setEdit(undefined);
+    try {
+      const response = await fetch(
+        `${await discoveryApi.getBaseUrl('todolist')}/todos`,
+        {
+          method: 'PUT',
+          body: JSON.stringify({ title: todo.title, id: todo.id }),
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        },
+      );
+      if (!response.ok) {
+        const { error } = await response.json();
+        alertApi.post({
+          message: error.message,
+          severity: 'error',
+        });
+        return;
+      }
+      refetchTodos();
+    } catch (e: any) {
+      alertApi.post({ message: e.message, severity: 'error' });
+    }
+  };
+
+  return (
+    <Page themeId="tool">
+      <Header
+        title="Welcome to todo-list!"
+        subtitle="Just a CRU todo list plugin"
+      >
+        <HeaderLabel label="Owner" value="Team X" />
+        <HeaderLabel label="Lifecycle" value="Alpha" />
+      </Header>
+      <Content>
+        <ContentHeader title="Todo List">
+          <SupportButton>A description of your plugin goes here.</SupportButton>
+        </ContentHeader>
+        <Grid container spacing={3} direction="column">
+          <Grid item>
+            <Typography variant="body1">Add todo</Typography>
+            <Box
+              component="span"
+              alignItems="flex-end"
+              display="flex"
+              flexDirection="row"
+            >
+              <TextField
+                placeholder="Write something here..."
+                onChange={e => (title.current = e.target.value)}
+              />
+              <Button variant="contained" onClick={handleAdd}>
+                Add
+              </Button>
+            </Box>
+          </Grid>
+          <Grid item>
+            <TodoList key={key} onEdit={setEdit} />
+          </Grid>
+        </Grid>
+      </Content>
+      {!!editElement && (
+        <EditModal
+          todo={editElement}
+          onSubmit={handleEdit}
+          onCancel={() => setEdit(undefined)}
+        />
+      )}
+    </Page>
+  );
+};
+
+function EditModal({
+  todo,
+  onCancel,
+  onSubmit,
+}: {
```

Comment:
This is a nit and veers a bit into matters-of-taste territory but, since this is an example after all - we should stop unpacking props like this, I think. Even for unexported components. Set the habits, as it were.

Reply:
is there any particular benefit even for un-exported components?

Reply:
but this isn't a massive object, it contains 3 objects. So, if you hover on the component, you see 3 props immediately so one doesn't have to click and navigate somewhere else to understand how the component works. So I guess it depends on the case 😄